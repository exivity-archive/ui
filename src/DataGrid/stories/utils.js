/**
 * Generate a unique random 8 character id (not cryptographically secure).
 *
 * @returns {string}
 */
const randomIdMemory = []
export const randomId = () => {
  let id

  if (process.env.NODE_ENV !== 'test') {
    do {
      id = Math.random().toString(36).substring(2, 10)
    } while (randomIdMemory.indexOf(id) > -1)
  } else {
    id = randomIdMemory.length
  }

  randomIdMemory.push(id)

  return id
}

/* eslint-disable-next-line max-len */
const RANDOM_WORDS = ['abstrusity', 'advertisable', 'bellwood', 'benzole', 'boreum', 'brenda', 'cassiopeian', 'chansonnier', 'cleric', 'conclusional', 'conventicle', 'copalm', 'cornopion', 'crossbar', 'disputative', 'djilas', 'ebracteate', 'ephemerally', 'epidemical', 'evasive', 'eyeglasses', 'farragut', 'fenny', 'ferryman', 'fluently', 'foreigner', 'genseng', 'glaiket', 'haunch', 'histogeny', 'illocution', 'imprescriptible', 'inapproachable', 'incisory', 'intrusiveness', 'isoceraunic', 'japygid', 'juiciest', 'jump', 'kananga', 'leavening', 'legerdemain', 'licence', 'licia', 'luanda', 'malaga', 'mathewson', 'nonhumus', 'nonsailor', 'nummary', 'nyregyhza', 'onanist', 'opis', 'orphrey', 'paganising', 'pebbling', 'penchi', 'photopia', 'pinocle', 'principally', 'prosector.', 'radiosensitive', 'redbrick', 'reexposure', 'revived', 'subexternal', 'sukarnapura', 'supersphenoid', 'tabularizing', 'territorialism', 'tester', 'thalassography', 'tuberculise', 'uncranked', 'undersawyer', 'unimpartible', 'unsubdivided', 'untwining', 'unwaived', 'webfoot', 'wedeling', 'wellingborough', 'whiffet', 'whipstall', 'wot', 'yonkersite', 'zonary']

function createRandomizedItems (depth, flat = true, parent) {
  let items = []
  let randomName
  let randomNumber
  let numChildren
  let randomValue

  if (process.env.NODE_ENV !== 'test') {
    randomName = RANDOM_WORDS[Math.floor(Math.random() * RANDOM_WORDS.length)]
    randomNumber = Math.floor(Math.random() * 100)
    numChildren = (depth < 3 && flat === false) ? Math.floor(Math.random() * 5) : 0
    randomValue = parseInt(Math.random() * 10, 10)
  } else {
    randomName = 'Test name'
    randomNumber = 50
    numChildren = (depth < 3 && flat === false) ? 3 : 0
    randomValue = 5
  }

  const random = randomId()

  const item = {
    id: random,
    parent: parent || null,
    name: randomName,
    number: randomNumber,
    group: 'Group ' + randomName.substr(0, 1),
    value: randomValue
  }

  items.push(item)

  for (let i = 0; i < numChildren; i++) {
    items = items.concat(createRandomizedItems(depth + 1, flat, item.key))
  }

  return items
}

export function createRandomizedData (items = 100, flat = true) {
  let data = []

  for (let i = 0; i < items; i++) {
    data = data.concat(createRandomizedItems(0, flat))
  }

  return data
}
