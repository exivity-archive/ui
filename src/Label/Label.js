import PropTypes from 'prop-types'
import { styled, Label as BaseLabel } from 'reakit'

const Label = styled(BaseLabel)``

Label.propTypes = {
  ...Label.propTypes,
  children: PropTypes.node,

  secondary: PropTypes.bool
}

Label.defaultProps = {}

export default Label
