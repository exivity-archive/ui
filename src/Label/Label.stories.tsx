import React from 'react'
import { Label } from '.'

// @ts-ignore
import { storiesOf } from '@storybook/react'

export default storiesOf('atoms|Label', module)
  .add('default', () => <Label>Default label</Label>)
  .add('secondary', () => <Label secondary>Secondary label</Label>)
  .add('nested', () => <Label>Primary label<Label>Secondary label</Label></Label>)
