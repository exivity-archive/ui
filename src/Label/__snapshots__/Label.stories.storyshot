// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots atoms|Label default 1`] = `
.c0 {
  padding: 12px;
}

.c2 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
}

.c2 > .c1 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c3 > .c1:first-child {
  padding-bottom: 0.625em;
  margin-right: unset;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c3 > .c1:first-child > .c1 {
  padding-bottom: unset;
}

.c4 > .c1:first-child {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c4 > .c1:first-child > .c1 {
  padding-bottom: unset;
}

.c4 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25em;
}

.c5 > .c1:first-child {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: 20em;
  -ms-flex-preferred-size: 20em;
  flex-basis: 20em;
}

.c5 > .c1:first-child > .c1 {
  padding-bottom: unset;
}

.c5 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25em;
}

.c6 > .c1:first-child {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: 30em;
  -ms-flex-preferred-size: 30em;
  flex-basis: 30em;
}

.c6 > .c1:first-child > .c1 {
  padding-bottom: unset;
}

.c6 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25em;
}

<div
  className="c0"
>
  <label
    className="c1 c2"
  >
    Default label
  </label>
</div>
`;

exports[`Storyshots atoms|Label nested 1`] = `
.c0 {
  padding: 12px;
}

.c2 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
}

.c2 > .c1 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c3 > .c1 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c4 > .c1:first-child {
  padding-bottom: 0.625em;
  margin-right: unset;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c4 > .c1:first-child > .c1 {
  padding-bottom: unset;
}

.c5 > .c1:first-child {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c5 > .c1:first-child > .c1 {
  padding-bottom: unset;
}

.c5 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25em;
}

.c6 > .c1:first-child {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: 20em;
  -ms-flex-preferred-size: 20em;
  flex-basis: 20em;
}

.c6 > .c1:first-child > .c1 {
  padding-bottom: unset;
}

.c6 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25em;
}

.c7 > .c1:first-child {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: 30em;
  -ms-flex-preferred-size: 30em;
  flex-basis: 30em;
}

.c7 > .c1:first-child > .c1 {
  padding-bottom: unset;
}

.c7 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25em;
}

<div
  className="c0"
>
  <label
    className="c1 c2"
  >
    Primary label
    <label
      className="c1 c2"
    >
      Secondary label
    </label>
  </label>
</div>
`;

exports[`Storyshots atoms|Label secondary 1`] = `
.c0 {
  padding: 12px;
}

.c3 > .c1 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c2 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c2 > .c1 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c4 > .c1:first-child {
  padding-bottom: 0.625em;
  margin-right: unset;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c4 > .c1:first-child > .c1 {
  padding-bottom: unset;
}

.c5 > .c1:first-child {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c5 > .c1:first-child > .c1 {
  padding-bottom: unset;
}

.c5 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25em;
}

.c6 > .c1:first-child {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: 20em;
  -ms-flex-preferred-size: 20em;
  flex-basis: 20em;
}

.c6 > .c1:first-child > .c1 {
  padding-bottom: unset;
}

.c6 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25em;
}

.c7 > .c1:first-child {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: 30em;
  -ms-flex-preferred-size: 30em;
  flex-basis: 30em;
}

.c7 > .c1:first-child > .c1 {
  padding-bottom: unset;
}

.c7 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25em;
}

<div
  className="c0"
>
  <label
    className="c1 c2"
  >
    Secondary label
  </label>
</div>
`;
