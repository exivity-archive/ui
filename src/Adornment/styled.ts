import styled, { css } from 'styled-components'

import { SizesProps, matchThemeProp } from '../utils/styled'
import { Position } from './Adornment'
import { Block } from '../Block'

export interface StyledAdornmentProps {
  position: Position
  inset: number | string
}

export const StyledAdornment = styled.span <StyledAdornmentProps>`
  display: flex;
  align-items: center;
  position: absolute;
  height: 100%;
  top: 0;
  font-size: ${
  matchThemeProp(
    theme => theme.global.sizes,
    { modifier: (em: number) => em * 20 }
  )}px;

  ${props => props.position === Position.LEFT
    ? css`left: ${props.inset}px;`
    : css`right: ${props.inset}px;`
  }
`

export const AdornmentWrapper = styled(Block)`
  position: relative;
  display: inline-block;
`
