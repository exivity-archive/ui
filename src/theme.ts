import { DeepReadonly } from './utils/types'
import './assets/styles/fonts.css'

const theme = {
  fonts: {
    custom: {
      families: [
        'Fira Mono',
        'Pt Sans'
      ],
      urls: [
        './assets/styles/fonts.css'
      ]
    }
  },
  global: {
    margin: '20px !default',
    fontFamily: 'Pt Sans'
  },
  colors: {
    gray: '#aaa',
    black: '#000',
    darker: '#222',
    dark: '#2e2e2e',
    kindadarkish: '#444',
    darkish: '#777',
    grayish: '#ccc',
    silver: '#e2e2e2',
    silverish: '#eee',
    light: '#f4f4f4',
    white: '#fff',
    darkblue: '#438dc7',
    blue: '#00a8d8',
    lightblue: '#5cccea',
    lighterblue: '#ccf4ff',
    lightestblue: '#e6f9ff',
    red: '#c33c32',
    yellow: '#fff68c',
    green: '#238e47'
  }
}

export type Theme = DeepReadonly<typeof theme>

export default theme
