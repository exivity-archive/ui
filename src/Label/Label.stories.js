import React from 'react'
import { storiesOf } from '@storybook/react'

import Label from './Label'

storiesOf('atoms|Label', module)
  .add('default', () => <Label>I'm a label</Label>)
