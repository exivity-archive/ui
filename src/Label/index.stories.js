import React from 'react'
import Label from '../Label'

import { storiesOf } from '@storybook/react'
import { CenterDecorator } from '../utils/testing/decorators/CenterDecorator'

export default storiesOf('Label', module)
  .addDecorator(CenterDecorator)
  .add('Label', () => {
    return <Label name='Label name' description='Label description' />
  })
