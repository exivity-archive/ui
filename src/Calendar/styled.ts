import styled, { css } from 'styled-components'
import { fromTheme, globalFont } from '../utils/styled'

export const StyledWeekDays = styled.ul`
  ${globalFont}
  opacity: 0.6;
  margin: 0;
  padding: 10px 0;

  li {
    display: inline-block;
    width: 14.28%;
    text-align: center;
  }
`

export const StyledDays = styled.ul`
  ${globalFont}

  padding: 10px 0;
  margin: 0;

  li {
    list-style-type: none;
    display: inline-block;
    width: calc(14.28% - 5px);
    text-align: center;
    margin-right: 5px;
    margin-bottom: 5px;
  }
`

export const StyledDay = styled.div<{ active: boolean }>`
  height: 100%;
  border-radius: 180px;
  padding: 5px;
  cursor: pointer;
  user-select: none;

  &:hover {
    color: white;
    background-color: ${fromTheme(theme => theme.global.purposes.primary)};
  }

  ${({ active }) => active && css`
    color: white;
    background-color: ${fromTheme(theme => theme.global.purposes.primary)};
  `}
`
