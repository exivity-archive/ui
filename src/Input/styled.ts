import styled, { css } from 'styled-components'
import { textAlign, TextAlignProps } from 'styled-system'

import { globalFont, matchThemeProp, fromTheme, hexToString, PurposesProps, SizesProps, StyledProps } from '../utils/styled'
import { Icon } from '../Icon'
import { Block } from '../Block'
import { animated } from 'react-spring'

interface ContainerProps {
  inline?: boolean
}

export const StyledContainer = styled.div<ContainerProps>`
  position: relative;

  ${props => (props.inline) && css`
    display: inline;
  `}

  ${Icon} {
    pointer-events: none;
  }
`

export interface StyledInputProps extends PurposesProps, SizesProps, StyledProps, TextAlignProps {
  // Variants
  outlined?: boolean
  flat?: boolean

  // Layout
  inline?: boolean
}

export const inputStyles = css<StyledInputProps>`
  ${globalFont};
  ${textAlign};

  font-size: ${matchThemeProp(theme => theme.global.sizes)}rem;

  box-sizing: border-box;
  padding: calc(0.5em - ${fromTheme(theme => theme.global.borderWidth)}px) 0.5em; // subtract border to get a height of exactly 2.5em for single line items

  border-radius: ${fromTheme(theme => theme.global.borderRadius)}px;
  outline: 0;
  border: 0;

  --focus-color: ${matchThemeProp(theme => theme.global.purposes, { modifier: hexToString })};

  ${props => (!props.inline) && css`
    display: block;
    width: 100%;
  `}

  ${props => (!props.outlined && !props.flat) && css`
    border: ${fromTheme(theme => theme.global.borderWidth)}px solid ${fromTheme(theme => theme.colors.lightGray)};
    background-color: ${fromTheme(theme => theme.colors.lightGray)};

    &:hover {
      border-bottom: ${fromTheme(theme => theme.global.borderWidth)}px solid rgba(var(--focus-color), 0.5);
    }

    &:focus {
      border-bottom: ${fromTheme(theme => theme.global.borderWidth)}px solid rgba(var(--focus-color), 1);
    }
  `}

  ${props => (props.outlined && !props.flat) && css`
    border: ${fromTheme(theme => theme.global.borderWidth)}px solid ${matchThemeProp(theme => theme.global.purposes)};
    background-color: unset;

    &:hover {
      border: ${fromTheme(theme => theme.global.borderWidth)}px solid ${fromTheme(theme => theme.colors.gray)};
    }

    &:focus {
      border: ${fromTheme(theme => theme.global.borderWidth)}px solid ${fromTheme(theme => theme.colors.dark)};
    }
  `}

  ${props => props.flat && css<StyledInputProps>`
    padding: 0;

    ${props => props.outlined && css`
      outline-offset: 9px;

      &:hover {
        outline: ${fromTheme(theme => theme.global.borderWidth)}px solid rgba(var(--focus-color), 0.5);
      }

      &:focus {
        outline: ${fromTheme(theme => theme.global.borderWidth)}px solid rgba(var(--focus-color), 1);
      }
    `}
  `}

  &::placeholder {
    color: currentcolor;
    opacity: 0.5;
  }

  &[disabled] {
    cursor: not-allowed;
    box-shadow: inset 0 0 999em rgba(128, 128, 128, 0.2);
  }
`

export const StyledInput = styled.input<StyledInputProps>`
  ${inputStyles}
`

export const AnimatedStyledInput = animated(StyledInput)
