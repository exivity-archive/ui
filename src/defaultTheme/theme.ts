import { preciseEm } from '../utils/styled/isolated'
import { DeepReadonly } from '../utils/types'
import { StyledProps } from 'styled-components'
import { ThemeResolver } from '../utils/styled'

export const BASE_SIZE = 16

const palette = {
  black: '#000',
  dark: '#222',
  gray: '#999',
  lightGray: '#eee',
  white: '#fff',
  blue: '#00a8d8',
  red: '#ef5350',
  yellow: '#fff68c',
  green: '#4caf50'
}

const theme = {
  fonts: {
    custom: {
      families: [],
      urls: [
        // Optional
        // 'font/stylesheet.css'
      ]
    },
    google: {
      families: [
        'Fira Mono:400',
        'Fira Sans:400,600'
      ]
    }
  },
  global: {
    baseSize: BASE_SIZE,
    fontFamily: `'Fira Sans', sans-serif`,
    fontFamilyMonospace: `'Fira Mono', monospace`,
    textColor: palette.dark,
    textColorMuted: palette.gray,
    lineHeight: 1.5,
    borderRadius: 4, // px
    borderWidth: 1, // px
    outlineWidth: 4, // px
    outlineAlpha: 0.15,
    shadowAlpha: 0.1,
    spacing: preciseEm(1.25), // 20px
    purposes: {
      primary: palette.blue,
      secondary: palette.gray,
      success: palette.green,
      danger: palette.red
    },
    sizes: {
      small: preciseEm(0.85), // 14px
      large: preciseEm(1.2) // 18px
    }
  },
  colours: {
    ...palette
  }
}

export type Theme = DeepReadonly<typeof theme>

export const fromTheme = (themeResolver: ThemeResolver) => (theme: Theme) => {
  return themeResolver(theme)
}

export default theme
