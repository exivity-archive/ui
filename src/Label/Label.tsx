import React from 'react'
import styled, { css } from 'styled-components'
import defaultStyledProps from '../utils/testing/defaultStyledProps'
import { fromTheme, globalFont, StyledProps } from '../utils/theme'

interface LabelProps extends StyledProps {
  secondary?: boolean
}

const secondary = css`
  display: block;
  font-size: ${fromTheme(theme => theme.global.sizes.small)}em;
  color: ${fromTheme(theme => theme.global.textColorMuted)};
`

const Label = styled.label <LabelProps>`
  ${globalFont}

  ${props => props.secondary && secondary}

  & > & {
    ${secondary}
  }
`

Label.defaultProps = defaultStyledProps

export default Label
