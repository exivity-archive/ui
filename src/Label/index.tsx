import * as React from 'react'
import styled from 'styled-components'
import theme, { Theme } from '../theme'

const SubLabelWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 33%;
  min-width: 250px;
`

const TopLabel = styled.span`
  font-weight: 600;
  color: #444444;
`
interface ISublabelProps {
  theme: Theme
}

const SubLabel = styled.div`
  margin-top: 5px;
  font-size: 12px;
  color: ${(p: ISublabelProps) => p.theme.colors.gray};
`

const LabelContent = styled.div`
  padding-left: 40px;
`

interface ILabelProps {
  name: string
  description?: string
  className?: string
}

const Label: React.FC<ILabelProps> = ({ name, description, children, className }) => (
  <div className={className}>
    <SubLabelWrapper>
      <TopLabel>{name}</TopLabel>
      <SubLabel theme={theme}>{description}</SubLabel>
    </SubLabelWrapper>
    <LabelContent>
    {children}
    </LabelContent>
  </div>
)

export default styled(Label)`
  display: flex;
  flex-direction: row;
  align-items: center;
  flex-wrap: wrap;
  width: 100%;
  margin: 40px 0;
`
