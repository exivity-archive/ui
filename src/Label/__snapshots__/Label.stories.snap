// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots atoms|Label default 1`] = `
.c0 {
  padding: 12px;
}

.c2 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
}

.c2 > .c1 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c3 .c1 {
  padding-bottom: 0.625em;
  margin-right: unset;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c3 .c1 > .c1 {
  padding-bottom: unset;
}

.c4 .c1 {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c4 .c1 > .c1 {
  padding-bottom: unset;
}

.c4 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25;
}

.c5 .c1 {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-flex-basis: 20em;
  -ms-flex-preferred-size: 20em;
  flex-basis: 20em;
}

.c5 .c1 > .c1 {
  padding-bottom: unset;
}

.c5 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25;
}

.c6 .c1 {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-flex-basis: 40%;
  -ms-flex-preferred-size: 40%;
  flex-basis: 40%;
}

.c6 .c1 > .c1 {
  padding-bottom: unset;
}

.c6 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25;
}

<div
  className="c0"
>
  <label
    className="c1 c2"
  >
    Default label
  </label>
</div>
`;

exports[`Storyshots atoms|Label nested 1`] = `
.c0 {
  padding: 12px;
}

.c2 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
}

.c2 > .c1 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c3 > .c1 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c4 .c1 {
  padding-bottom: 0.625em;
  margin-right: unset;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c4 .c1 > .c1 {
  padding-bottom: unset;
}

.c5 .c1 {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c5 .c1 > .c1 {
  padding-bottom: unset;
}

.c5 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25;
}

.c6 .c1 {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-flex-basis: 20em;
  -ms-flex-preferred-size: 20em;
  flex-basis: 20em;
}

.c6 .c1 > .c1 {
  padding-bottom: unset;
}

.c6 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25;
}

.c7 .c1 {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-flex-basis: 40%;
  -ms-flex-preferred-size: 40%;
  flex-basis: 40%;
}

.c7 .c1 > .c1 {
  padding-bottom: unset;
}

.c7 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25;
}

<div
  className="c0"
>
  <label
    className="c1 c2"
  >
    Primary label
    <label
      className="c1 c2"
    >
      Secondary label
    </label>
  </label>
</div>
`;

exports[`Storyshots atoms|Label secondary 1`] = `
.c0 {
  padding: 12px;
}

.c3 > .c1 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c2 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c2 > .c1 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c4 .c1 {
  padding-bottom: 0.625em;
  margin-right: unset;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c4 .c1 > .c1 {
  padding-bottom: unset;
}

.c5 .c1 {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c5 .c1 > .c1 {
  padding-bottom: unset;
}

.c5 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25;
}

.c6 .c1 {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-flex-basis: 20em;
  -ms-flex-preferred-size: 20em;
  flex-basis: 20em;
}

.c6 .c1 > .c1 {
  padding-bottom: unset;
}

.c6 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25;
}

.c7 .c1 {
  padding-bottom: unset;
  margin-right: 1.25em;
  -webkit-flex-basis: 40%;
  -ms-flex-preferred-size: 40%;
  flex-basis: 40%;
}

.c7 .c1 > .c1 {
  padding-bottom: unset;
}

.c7 > *:not(.c1):not(:last-child) {
  margin-bottom: 1.25;
}

<div
  className="c0"
>
  <label
    className="c1 c2"
  >
    Secondary label
  </label>
</div>
`;
